pkg/pdatatest/pmetrictest/options.go

```diff
@@ -159,6 +159,64 @@ func roundDataPointSliceValues(dataPoints pmetric.NumberDataPointSlice, precisio
 	}
 }
 
+// IgnoreExemplars is a CompareMetricsOption that clears exemplar fields on all the Sum data points.
+func IgnoreExemplars() CompareMetricsOption {
+	return compareMetricsOptionFunc(func(expected, actual pmetric.Metrics) {
+		maskExemplars(expected)
+		maskExemplars(actual)
+	})
+}
+
+func maskExemplars(metrics pmetric.Metrics) {
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		ilms := rms.At(i).ScopeMetrics()
```

Comment:
Example of using new iterators.

```suggestion
	for _, ilms := range rms.All() {
```

Reply:
I don't think this is viable in this particular use-case, since range operator creates a copy of an element, which we are trying to modify in our function and therefore the change is not happening in the original element in the slice, but just in it's copy which results in no modification. I guess the `.All()` function is meant for reading data than modifying it.